core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -110,10 +112,9 @@
     return this;
   }
 
-  @Override
-  public void start(Listener<RespT> observer, Metadata headers) {
-    Preconditions.checkState(stream == null, "Already started");
-
+  @VisibleForTesting
+  static void prepareHeaders(Metadata headers, CallOptions callOptions, String userAgent,
```

Comment:
My rule of thumb is to consolidate code after there are at least 3 implementations, so that the common and reusable parts are obvious.  Considering there are no other users of this yet, I am reluctant to make it reusable (not that it isn't a good idea, I just don't have confidence yet).